Lib/email/_header_value_parser.py

```diff
@@ -1744,7 +1744,10 @@ def get_name_addr(value):
             raise errors.HeaderParseError(
                 "expected name-addr but found '{}'".format(token))
         if leader is not None:
-            token[0][:0] = [leader]
+            if isinstance(token[0], TokenList):
```

Comment:
This part of the fix doesn't seem to be related to the linked bpo, can you please explain the need for this?

I see that both the test cases are passing without this change.

Reply:
The second case fails without this?
```
Run tests in parallel using 3 child processes
0:00:11 load avg: 2.76 [1/1/1] test_email failed
test test_email failed -- Traceback (most recent call last):
  File "/home/bb/tmp/cpython/Lib/email/_header_value_parser.py", line 1951, in get_address
    token, value = get_group(value)
  File "/home/bb/tmp/cpython/Lib/email/_header_value_parser.py", line 1909, in get_group
    raise errors.HeaderParseError("expected ':' at end of group "
email.errors.HeaderParseError: expected ':' at end of group display name but found '<jxd@example.com>'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/bb/tmp/cpython/Lib/test/test_email/test_headerregistry.py", line 1286, in test_complex_address_list
    h = self.make_header('to', source)
  File "/home/bb/tmp/cpython/Lib/test/test_email/test_headerregistry.py", line 79, in make_header
    return self.factory(name, value)
  File "/home/bb/tmp/cpython/Lib/email/headerregistry.py", line 602, in __call__
    return self[name](name, value)
  File "/home/bb/tmp/cpython/Lib/email/headerregistry.py", line 197, in __new__
    cls.parse(value, kwds)
  File "/home/bb/tmp/cpython/Lib/email/headerregistry.py", line 340, in parse
    kwds['parse_tree'] = address_list = cls.value_parser(value)
  File "/home/bb/tmp/cpython/Lib/email/headerregistry.py", line 331, in value_parser
    address_list, value = parser.get_address_list(value)
  File "/home/bb/tmp/cpython/Lib/email/_header_value_parser.py", line 1974, in get_address_list
    token, value = get_address(value)
  File "/home/bb/tmp/cpython/Lib/email/_header_value_parser.py", line 1954, in get_address
    token, value = get_mailbox(value)
  File "/home/bb/tmp/cpython/Lib/email/_header_value_parser.py", line 1777, in get_mailbox
    token, value = get_name_addr(value)
  File "/home/bb/tmp/cpython/Lib/email/_header_value_parser.py", line 1760, in get_name_addr
    token[0][:0] = [leader]
TypeError: 'ValueTerminal' object does not support item assignment

== Tests result: FAILURE ==

1 test failed:
    test_email

Total duration: 11 sec 830 ms
Tests result: FAILURE
```

Reply:
Ah, okay! Thanks!